src/schedule/zephyr_dp_schedule.c

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Szkudlinski
+ */
+
+#include <sof/audio/component.h>
+#include <rtos/task.h>
+#include <stdint.h>
+#include <sof/schedule/dp_schedule.h>
+#include <sof/schedule/ll_schedule_domain.h>
+#include <sof/trace/trace.h>
+#include <rtos/wait.h>
+#include <rtos/interrupt.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys_clock.h>
+
+#include <zephyr/kernel/thread.h>
+
+LOG_MODULE_REGISTER(dp_schedule, CONFIG_SOF_LOG_LEVEL);
+/* 87858bc2-baa9-40b6-8e4c-2c95ba8b1545 */
+DECLARE_SOF_UUID("dp-schedule", dp_sched_uuid, 0x87858bc2, 0xbaa9, 0x40b6,
+		 0x8e, 0x4c, 0x2c, 0x95, 0xba, 0x8b, 0x15, 0x45);
+
+DECLARE_TR_CTX(dp_tr, SOF_UUID(dp_sched_uuid), LOG_LEVEL_INFO);
+
+struct scheduler_dp_data {
+	struct list_item tasks;		/* list of active dp tasks */
+	struct k_spinlock lock;		/* synchronization between cores */
+};
+
+struct task_dp_pdata {
+	k_tid_t thread_id;		/* zephyr thread ID */
+	k_thread_stack_t *p_stack;	/* pointer to thread stack */
```

Comment:
also as a further enhancement - we could try explicitly declare the stack pointer as `__sparse_cache` since stack should be cached.